### PR TITLE
Move tool event functions out of top toolbar

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
This PR moves the terraform tool event functions out of the top toolbar window, reducing the complexity considerably.

- [x] Might also be able to localise a couple of globals as a result of this.
- [x] Fix windowless custom tools defined in user scripts.
- [x] Needs some thorough testing of the terraform and scenery tools.
- [x] Take advantage of reduced complexity to fix 'toolbarless' terraform shortcuts.

### CODE REVIEW

**Comment by karst** [comment]
image
I have not noticed any differences in behaviour

**Comment by AaronVanGeffen** [comment]
Looks like custom tools (i.e. those from scripts) may be broken by this, particularly those that are active without a visible window. Maybe `CustomWindow.cpp` isn't the best location for firing their events, then. Still, neither is `TopToolbar.cpp`. Needs to be fixed before this is merged, though, clearly.

**Comment by AaronVanGeffen** [comment]
Tried fiddling with it, but moving the script tool events into `MouseInput.cpp` wasn't viable either, nor exactly clean. Particularly problematic is that `ToolCancel` is not used exclusively in the UI subproject, so it doesn't have access to the custom tools from the script engine.

For now, I think it best to leave the custom tool events in the `TopToolbar.cpp` file. I have cleaned up the functions somewhat, though, and added a comment with the rationale for having them here.

File: src/openrct2/windows/_legacy.cpp [hunk 1]
```diff
@@ -415,8 +415,10 @@ bool SceneryToolIsActive()
     auto toolWindowClassification = gCurrentToolWidget.window_classification;
     WidgetIndex toolWidgetIndex = gCurrentToolWidget.widget_index;
     if (InputTestFlag(INPUT_FLAG_TOOL_ACTIVE))
-        if (toolWindowClassification == WindowClass::TopToolbar && toolWidgetIndex == WC_TOP_TOOLBAR__WIDX_SCENERY)
+    {
+        if (toolWindowClassification == WindowClass::Scenery && toolWidgetIndex == 0) // TODO: global widx
```
Comment (Gymnasiast) [inline src/openrct2/windows/_legacy.cpp:419]: TODO still needs addressing.

**Review by Gymnasiast — APPROVED** [review]
Testing revealed no bugs.

**Pull request merged**